### PR TITLE
Cinnamon 5.6 updates 11/26

### PR DESCRIPTION
**Pull request by SparkyBluefang**

### CODE REVIEW

**Comment by ceamac** [comment]
I ran `qa-vdb` from `app-portage/iwdevtools`.  Please check if they are required at runtime:
```
 * VDB: detected possibly incorrect RDEPEND (gnome-extra/nemo-5.6.0-r1)
 * dev-libs/gobject-introspection:= < 
 * gnome-base/dconf:=               < 
```

**Comment by ceamac** [comment]
```
 * VDB: detected possibly incorrect RDEPEND (x11-wm/muffin-5.6.1)
 * dev-libs/gobject-introspection:= < 
 *                                  > media-libs/fontconfig
 * media-libs/mesa                  < 
 * sys-apps/dbus                    < 
 * x11-libs/cairo:=                 | x11-libs/cairo
 * x11-libs/libXau                  < 
 * x11-libs/libXrender              < 
 * x11-libs/startup-notification:=  | x11-libs/startup-notification
```

**Comment by ceamac** [comment]
With USE="-eds -gstreamer -gtk-doc -networkmanager -nls"
```
 * VDB: detected possibly incorrect RDEPEND (gnome-extra/cinnamon-5.6.4)
 * app-accessibility/caribou           < 
 * dev-libs/gobject-introspection:=    | dev-libs/gobject-introspection
 * dev-libs/keybinder:3                < 
 * dev-libs/libtimezonemap             < 
 * gnome-base/dconf                    < 
 * gnome-base/libgnomekbd              < 
 * gnome-extra/cinnamon-control-center < 
 * gnome-extra/cinnamon-desktop:=      < 
 * gnome-extra/cinnamon-screensaver    < 
 * gnome-extra/nemo                    < 
 * media-libs/gsound                   < 
 * net-libs/libsoup:2.4                < 
 * sys-apps/accountsservice            < 
 * sys-apps/dbus                       < 
 * sys-apps/pciutils                   < 
 * sys-apps/util-linux                 < 
 * sys-auth/polkit                     < 
 * sys-power/upower                    < 
 * virtual/opengl                      < 
 *                                     > x11-libs/cairo
 * x11-libs/libnotify:=                < 
```

**Comment by ceamac** [comment]
USE="-systemd -xinerama"
```
VDB: detected possibly incorrect RDEPEND (gnome-extra/cinnamon-screensaver-5.6.2)
app-accessibility/caribou      < 
gnome-extra/cinnamon-desktop:= < 
sys-apps/accountsservice       < 
sys-apps/dbus                  < 
sys-auth/elogind               < 
                               > x11-libs/cairo
x11-libs/libXrandr             < 
                               > x11-libs/pango
```

**Comment by ceamac** [comment]
A little explanation about how to interpret the output of qa-vdb:
- the left side contains packages already in RDEPEND that are not linked against directly.  They may be needed, or they may not.  You should check for dlopen for example, see if they really should be there.
- the right side contains packages that are linked against specifically, but do not appear in RDEPEND.  These must be fixed, they need to be added to RDEPEND, they are indeed used.
- the case where both sides are present is self-explanatory, you need to add or remove a slot operator.

File: dev-python/python3-xapp/python3-xapp-2.4.0-r1.ebuild [hunk 1]
```diff
@@ -44,6 +44,10 @@ src_compile() {
 	python_foreach_impl meson_src_compile
 }
```
Comment (ceamac) [inline dev-python/python3-xapp/python3-xapp-2.4.0-r1.ebuild:45]: I suggest adding a `|| die` on line 30 (after the `echo` in src_prepare)

**Review by ceamac — COMMENTED** [review]
Let's change `PYTHON_COMPAT=( python3_{8,9,10,11} )` to `PYTHON_COMPAT=( python3_{8..11} )`

Thanks!

**Comment by thesamesam** [comment]
It is based on whether the dependency is linked against and exposes a subslot.

**Comment by ceamac** [comment]
Transitive dependencies are only linked if they are used, because `LDFLAGS` contains `-Wl,--as-needed`.  So if you see them on the right side, they are needed, even if meson does not list them explicitly.  You can check that with `lddtree` from `app-misc/pax-utils`, which can show direct dependencies and dependencies of dependencies.

**Comment by thesamesam** [comment]
I'll merge it as it is so we can get the fixes in, as the general dependency checking stuff can be applied going forward.

**Review by ceamac — APPROVED** [review]

**Comment by SparkyBluefang** [comment]
:+1: I've been otherwise occupied with a whole lot of year-end work and holiday stuff. I'll go through and fix-up all of the Cinnamon packages in my next update batch.

**Pull request closed without merging**